### PR TITLE
Add jdbc-shaded profile to support IDE debug

### PR DESCRIPTION
**Pull request by cxzl25**
### _Why are the changes needed?_
1. `mvn antrun:run@build-info -pl kyuubi-common`
2. debug engine suite

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by pan3793** [comment]
How about using `kyuubi-hive-jdbc` in default and switch to `kyuubi-hive-jdbc-shaded` in CI?

File: kyuubi-common/src/main/scala/org/apache/kyuubi/package.scala [hunk 1]
```diff
@@ -30,7 +30,8 @@ package object kyuubi {
     if (buildFileStream == null) {
       throw new KyuubiException(s"Can not load the core build file: $buildFile, if you meet " +
         s"this exception when running unit tests " +
-        s"please make sure you have run the `mvn package` or `mvn generate-resources` command.")
+        s"please make sure you have run the `mvn package` or " +
+        s"`mvn antrun:run@build-info -pl kyuubi-common` command.")
```
Comment (cxzl25) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/package.scala:34]:
```
Caused by: org.apache.kyuubi.KyuubiException: Can not load the core build file: kyuubi-version-info.properties, if you meet this exception when running unit tests please make sure you have run the `mvn package` or `mvn antrun:run@build-info -pl kyuubi-common` command.
	at org.apache.kyuubi.package$BuildInfo$.<init>(package.scala:31)
	at org.apache.kyuubi.package$BuildInfo$.<clinit>(package.scala)
```

Comment (cxzl25) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/package.scala:34]: It seems that no matter what phase the `maven-antrun-plugin` is configured in, IntelliJ IDEA will not run when building the project, so the `kyuubi-version-info.properties` file will not be generated.

File: pom.xml [hunk 2]
```diff
@@ -2332,6 +2334,13 @@
             </properties>
         </profile>
 
+        <profile>
+            <id>jdbc-use-shaded</id>
```
Comment (pan3793) [inline pom.xml:2338]: jdbc-shaded

**Review by pan3793 — APPROVED** [review]

File: kyuubi-common/src/main/scala/org/apache/kyuubi/package.scala (hunk 1 above)
Comment (cxzl25) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/package.scala:34]: If you specify the execution id to run ( `mvn antrun:run@build-info -pl kyuubi-common` ) , you need maven version 3.3.1 or higher.

> MNG-5768 specify execution-id for direct plugin goal invocation from command line
https://issues.apache.org/jira/browse/MNG-5768

**Pull request closed without merging**